language/move-prover/bytecode/src/read_write_set_analysis.rs

```diff
@@ -879,3 +842,67 @@ impl Default for ReadWriteSetState {
         }
     }
 }
+
+// =================================================================================================
+// Converting Infer Result into standalone types
+
+impl Offset {
+    pub fn normalize(&self, env: &GlobalEnv) -> Option<RWOffset> {
+        Some(match self {
+            Offset::Field(idx) => RWOffset::Field(*idx),
+            Offset::VectorIndex => RWOffset::VectorIndex,
+            Offset::Global(s) => RWOffset::Global(s.get_type().into_struct_type(env)?),
+        })
+    }
+}
+impl AccessPath {
+    pub fn normalize(&self, env: &GlobalEnv) -> Option<Vec<RWAccessPath>> {
```

Comment:
What is the difference in meaning between `None` and `Some(vec![])` here?

Reply:
It will return `None` when there's any malformed AccessPath, e.g: type mismatch. It should never return `Some(vec![])` because the two branches that leads to some will at least have one element in the root.

Reply:
Got it. In that case, I think I'd `expect` when you see a malformed path (it's not like the caller can recover from that, and we don't want them to proceed with the incorrect result) + just return `Vec<RWAccessPath>`.

Reply:
Done!